### ISSUE TITLE
`solc` version outdated

### ISSUE THREAD

**Issue by aaronzshey** [NomicFoundation/hardhat#5427]
## The Problem

The version of `solc` hardhat uses is 0.7.3, released 4 years ago.  

Why should we fix it though?

When I install `hardhat`, I get the following warns: 

```
npm WARN deprecated rimraf@2.7.1: Rimraf versions prior to v4 are no longer supported
npm WARN deprecated glob@7.2.0: Glob versions prior to v9 are no longer supported
npm WARN deprecated glob@8.1.0: Glob versions prior to v9 are no longer supported
```

Here's the output of `pnpm why rimraf` 

```
devDependencies:
hardhat link:packages/hardhat-core
└─┬ solc 0.7.3
  └─┬ fs-extra 0.30.0
    └── rimraf 2.7.1
```

and `pnpm why glob`

```
devDependencies:
hardhat link:packages/hardhat-core
├── glob 7.2.0
├─┬ mocha 10.4.0
│ └── glob 8.1.0
└─┬ solc 0.7.3
  └─┬ fs-extra 0.30.0
    └─┬ rimraf 2.7.1
      └── glob 7.2.0
```

By upgrading solc from 0.7.3 to 0.8.26, we can eliminate outdated versions of glob and rimraf from the dependencies.  solc 0.8.26 has [7 dependencies](https://www.npmjs.com/package/solc/v/0.8.26?activeTab=dependencies), while solc 0.7.3 has [9 dependencies](https://www.npmjs.com/package/solc/v/0.7.3?activeTab=dependencies) (including the two aforementioned outdated ones)

## Steps to Reproduce the Problem

  1. create a new node project 
  2. install hardhat (`npm i hardhat`)
  3. read following warnings

## Specifications

  - Hardhat Version: 2.22.5
  - pnpm Version: 9.4.0
  - node version: 21.2.0
  - Platform: macOS 12.7.2

## More Info 

I have a huge pet peeve for npm warns!  I'd be super happy to be assigned to this issue and work on upgrading the solc version.  I'll link my fork in a second.  

## TL;DR:
solc is outdated; upgrading it will reduce warn messages on npm install and also hopefully reduce dependencies.  quality of life pretty much.
